src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs

```diff
@@ -0,0 +1,803 @@
+
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).ResponseTypes.Should().HaveCount(3);
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).ResponseTypes.ToArray()[0].Should().Be(OAuthResponseType.Token);
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).ResponseTypes.ToArray()[1].Should().Be(OAuthResponseType.IdToken);
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).ResponseTypes.ToArray()[1].Should().Be(OAuthResponseType.Code);
+
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).GrantTypes.Should().HaveCount(2);
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).GrantTypes.First().Should().Be(OAuthGrantType.Implicit);
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).GrantTypes.Last().Should().Be(OAuthGrantType.AuthorizationCode);
+
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).ApplicationType.Should().Be(OpenIdConnectApplicationType.Native);
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).TosUri.Should().Be("https://example.com/client/tos");
+                ((OpenIdConnectApplicationSettingsClient)retrievedSettings.OauthClient).PolicyUri.Should().Be("https://example.com/client/policy");
+            }
+            finally
+            {
+                await client.Applications.DeactivateApplicationAsync(createdApp.Id);
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        [Fact]
+        public async Task GetApplication()
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new BasicAuthApplication
+            {
+                Name = "template_basic_auth",
+                Label = "Sample Basic Auth App",
+                SignOnMode = ApplicationSignOnMode.BasicAuth,
+                Settings = new BasicApplicationSettings
+                {
+                    App = new BasicApplicationSettingsApplication
+                    {
+                        Url = "https://example.com/login.html",
+                        AuthUrl = "https://example.com/auth.html",
+                    },
+                },
+            });
+
+            try
+            {
+                var retrievedById = await client.Applications.GetApplicationAsync(createdApp.Id);
+                retrievedById.Id.Should().Be(createdApp.Id);
+            }
+            finally
+            {
+                await client.Applications.DeactivateApplicationAsync(createdApp.Id);
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        [Fact]
+        public async Task ListApplications()
```

Comment:
Add another test very similar to this one: `ListApplicationsOfType`

* Add two applications (of different types)
* Make sure LINQ `OfType` works, like this: https://github.com/okta/okta-sdk-dotnet/blob/master/src/Okta.Sdk.IntegrationTests/FactorScenarios.cs#L50